### PR TITLE
[fix] - prove ownership of registry and artifact locks

### PR DESCRIPTION
**Pull request by cgfixit**
## Branch naming
- Driver: Kimi / Kimi Code
- Branch: `kimi/cyclaw-optimize-agentic-lock-ownership`

---

## Proposed changes

Add process-ownership tokens to the two cross-process atomic-mkdir mutexes in the agentic layer, plus a test-import cleanup:

- `agentic/registry.py` — `_acquire_registry_lock()` now writes a `{pid, started_at}` JSON token into the lock directory immediately after creating it. `_release_registry_lock()` only removes the directory when the token names the current PID. `_can_reclaim_lock()` only allows reclaim when the recorded PID is dead, or when no token is present for backward compatibility.
- `agentic/harness_optimizer/patching.py` — reuses the same token helpers so `_acquire_artifact_lock()` / `_release_artifact_lock()` get the same ownership semantics.
- `tests/test_agentic_harness_phase679.py` — updated the `_LOCK_STALE_SEC` import from `agentic.harness_optimizer.patching` to `agentic.registry` after the constant moved in the refactor above (fixes the broad matrix `ImportError` failure).

This closes a race where a slow process could release a lock that had been reclaimed by another live process, and prevents a live-but-slow owner from losing its lock purely because the directory looked old.

### Invariant / Governance Impact

| Invariant | Impact | Evidence |
|-----------|--------|----------|
| I1 RAG-first | None | Core graph untouched |
| I2 Topology = policy | None | No routing changes |
| I3 Triple-gated external | None | No provider/client changes |
| I4 Audit convergence | Preserved | Existing `agentic_skill_applied` and `agentic_harness_candidate_accepted` audit events still fire |
| I5 Soul governance | None | Soul path untouched |
| I6 Module isolation | Preserved | agentic layer only; `gate.py`, `graph.py`, and `mcp_hybrid_server.py` are untouched |

No topology or governance changes.

---

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation Update
- [ ] Invariant / Governance refinement

**Scope note:** Out-of-band agentic layer (`agentic/registry.py`, `agentic/harness_optimizer/patching.py`) + tests.

---

## Benefits / why

- Eliminates a real but narrow cross-process lock race in the skills registry and harness-optimizer artifact apply paths.
- Keeps the existing zero-dependency `Path.mkdir()` mutex; no new imports beyond `shutil` already used elsewhere.
- Backward-compatible with token-less locks created by older code or by tests that create the directory directly.
- Restores CI collection for `tests/test_agentic_harness_phase679.py` across the full matrix.

---

## Risks to monitor

- PID reuse: the ownership check is best-effort via `os.kill(pid, 0)`. On a busy host a reused PID could masquerade as the original owner. The residual risk is bounded by the short apply duration and the 60-second stale window.
- Windows process-lifetime edge cases: `os.kill(0)` behavior is portable, but tests assert the dead-PID and live-PID paths explicitly.
- If a future change writes extra files into the lock directory, `_release_*_lock` will need to clean them before `rmdir()`.

---

## Checklist

- [x] Read latest `docs/CyClaw Architecture Guide` and `SECURITY.md`.
- [x] Preserves all 6 security invariants and I6 module isolation (OOB layer only).
- [x] Sandbox validation run:
  ```bash
  GROK_API_KEY=dummy pytest tests/test_agentic_harness_phase679.py -q --tb=short
  GROK_API_KEY=dummy pytest tests/test_agentic_harness_optimizer.py tests/test_agentic_registry.py -q --tb=short
  ruff check --select E,F,I,B,C4,UP,S agentic/registry.py agentic/harness_optimizer/patching.py tests/test_agentic_registry.py tests/test_agentic_harness_optimizer.py tests/test_agentic_harness_phase679.py
  ```
- [x] No new external network dependencies or mandatory online LLM assumptions.
- [x] Commit message follows title prefix convention (`[fix] - ...`).

---

## Further comments

No change to the core RAG path, graph topology, or public API. The lock directory still lives next to the registry/artifact file and is still removed on a clean apply. The only behavioral change is that release and reclaim now respect process identity, reducing the chance of one agentic process clobbering another's critical section. A follow-up commit fixes the stale test import that caused the broad matrix failure after `_LOCK_STALE_SEC` moved to `agentic.registry`.

## Verify
- `GROK_API_KEY=dummy pytest tests/test_agentic_harness_phase679.py -q --tb=short` → 26 passed
- `GROK_API_KEY=dummy pytest tests/test_agentic_harness_optimizer.py tests/test_agentic_registry.py -q --tb=short` → 66 passed
- `ruff check --select E,F,I,B,C4,UP,S agentic/registry.py agentic/harness_optimizer/patching.py tests/test_agentic_registry.py tests/test_agentic_harness_optimizer.py tests/test_agentic_harness_phase679.py` → all passed
- Local `verify-ci-emulation.py` pre-push gate passed (invariant-guard, gate/harness runtime, pytest due-diligence)

## Merge order
Independent PR. Can merge in any order relative to the other `kimi/cyclaw-optimize-*` branches; no shared files.

## Base
`main`

### CODE REVIEW

**Comment by cgfixit** [comment]
@codex review

File: agentic/registry.py [hunk 1]
```diff
@@ -51,6 +51,74 @@ def _utcnow() -> str:
...
+            return data
+    except (OSError, json.JSONDecodeError):
+        pass
+    return None
+
+
+def _pid_alive(pid: int) -> bool:
+    """Best-effort check that ``pid`` still exists in this machine's process table."""
+    try:
+        os.kill(pid, 0)
```
[the remark this answers is not in the document]
Comment (cgfixit) [reply agentic/registry.py:89]: @codex apply fixes
